File: Visual Studio 2010/CSCustomIEContextMenu/NativeMethods/tagOleMenuGroupWidths.cs
/*************************************** 模块头*****************************\
* 模块名:  tagOleMenuGroupWidths.cs
* 项目名:  CSCustomIEContextMenu
* 版权 (c) Microsoft Corporation.
* 
* 类 tagOleMenuGroupWidths 被 IOleInPlaceFrame 使用.
* 
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.Runtime.InteropServices;

namespace CSCustomIEContextMenu.NativeMethods
{
    [StructLayout(LayoutKind.Sequential)]
    public sealed class tagOleMenuGroupWidths
    {
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 6)]
        public int[] widths = new int[6];
    }
}

File: Visual Studio 2010/VBCustomIEContextMenu/NativeMethods/tagOleMenuGroupWidths.vb
'********************************** 模块头 ******************************'
' 模块名:  tagOleMenuGroupWidths.vb
' 项目名:  VBCustomIEContextMenu
' 版权 (c) Microsoft Corporation.
' 
' 类 tagOleMenuGroupWidths 被 IOleInPlaceFrame 使用.
' 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*************************************************************************'

Imports System.Runtime.InteropServices

Namespace NativeMethods
    <StructLayout(LayoutKind.Sequential)>
    Public NotInheritable Class tagOleMenuGroupWidths
        <MarshalAs(UnmanagedType.ByValArray, SizeConst:=6)>
        Public widths(5) As Integer
    End Class
End Namespace
